drivers/accel/adxl367/iio_adxl367.c

```diff
@@ -0,0 +1,936 @@
+			return iio_format_value(buf, len, IIO_VAL_INT, 1, &val);
+
+		default:
+			return -EINVAL;;
+		}
+
+	default:
+		return -EINVAL;
+	}
+}
+
+/***************************************************************************//**
+ * @brief Handles the write request for calibbias attribute.
+ *
+ * @param dev     - The iio device structure.
+ * @param buf	  - Command buffer to be filled with the data to be written.
+ * @param len     - Length of the received command buffer in bytes.
+ * @param channel - Command channel info.
+ * @param priv    - Command attribute id.
+ *
+ * @return ret    - Result of the writing procedure.
+*******************************************************************************/
+static int adxl367_iio_write_calibbias(void *dev, char *buf, uint32_t len,
+				       const struct iio_ch_info *channel, intptr_t priv)
+{
+	int32_t ret;
+	int32_t val;
+	uint16_t calibbias;
+	struct adxl367_iio_dev *iio_adxl367;
+	struct adxl367_dev *adxl367;
+
+	if (!dev)
+		return -EINVAL;
+
+	iio_adxl367 = (struct adxl367_iio_dev *)dev;
+
+	if (!iio_adxl367->adxl367_dev)
+		return -EINVAL;
+
+	adxl367 = iio_adxl367->adxl367_dev;
+
+	switch (channel->type) {
+	case IIO_ACCEL:
+		iio_parse_value(buf, IIO_VAL_INT, &val, NULL);
+		// Change this value to two's complement with sign bit = BIT15
+		if(val < 0)
+			calibbias = NO_OS_BIT(15) | (~abs(val) + 1);
+		else
+			calibbias = val;
+
+		// Set the device in standby mode
+		ret = adxl367_set_power_mode(adxl367, ADXL367_OP_STANDBY);
+		if (ret)
+			return ret;
+
+		// Set the offset on the requested axis
+		switch (channel->ch_num) {
+		case 0:
+			ret = adxl367_set_offset(adxl367, calibbias, adxl367->y_offset,
+						 adxl367->z_offset);
+			break;
+		case 1:
+			ret = adxl367_set_offset(adxl367, adxl367->x_offset, calibbias,
+						 adxl367->z_offset);
+			break;
+		case 2:
+			ret = adxl367_set_offset(adxl367, adxl367->x_offset, adxl367->y_offset,
+						 calibbias);
+			break;
+
+		default:
+			ret = -EINVAL;
+			break;
+		}
+		if (ret)
+			return ret;
+
+		// Set the device in measuring mode
+		return adxl367_set_power_mode(adxl367, ADXL367_OP_MEASURE);
+
+	default:
+		return -EINVAL;
+	}
+}
+
+/***************************************************************************//**
+ * @brief Handles the read request for offset attribute.
+ *
+ * @param dev     - The iio device structure.
+ * @param buf	  - Command buffer to be filled with requested data.
+ * @param len     - Length of the received command buffer in bytes.
+ * @param channel - Command channel info.
+ * @param priv    - Command attribute id.
+ *
+ * @return ret    - Result of the reading procedure.
+ * 					In case of success, the size of the read data is returned.
+*******************************************************************************/
+static int adxl367_iio_read_offset(void *dev, char *buf, uint32_t len,
+				   const struct iio_ch_info *channel, intptr_t priv)
+{
+	int32_t vals[2];
```

Comment:
it does not look like you need an array here seeing how you are using only vals[0]